src/drawing.js

```diff
@@ -0,0 +1,26 @@
+export const drawingHooks = ['init', 'afterDraw', 'beforeDraw', 'beforeDrawLabel', 'afterDrawLabel'];
+
+export function init(element, options, properties) {
+  invokeHook(options.init, element, properties);
+}
+
+export function drawElement(chart, element, caller) {
+  if (element.options.drawTime === caller && invokeHook(element.options.beforeDraw, element)) {
+    element.draw(chart.ctx);
+    invokeHook(element.options.afterDraw, element);
+  }
+}
+
+export function drawLabelElement(chart, element, caller) {
+  if ('drawLabel' in element && element.options.label && (element.options.label.drawTime || element.options.drawTime) === caller && invokeHook(element.options.beforeDrawLabel, element)) {
+    element.drawLabel(chart.ctx, chart.chartArea);
+    invokeHook(element.options.afterDrawLabel, element);
+  }
+}
+
+function invokeHook(callback, element, properties) {
```

Comment:
I'd use the `callback` helper from chart.js instead.
I'd also use the same logic with chart.js, you'd return `false` from the callback to cancel.

Reply:
yes, many other things are missing. But I can continue working on that, slowly but surely